x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/hooks/use_columns/use_columns.ts

```diff
@@ -29,54 +29,76 @@ const fieldTypeToDataGridColumnTypeMapper = (fieldType: string | undefined) => {
   return fieldType;
 };
 
+const getFieldCategoryFromColumnId = (columnId: string): string => {
+  const fieldName = columnId.split('.');
+
+  if (fieldName.length === 1) {
+    return 'base';
+  }
+
+  return fieldName[0];
+};
```

Comment:
I love this one, it simplifies the search to constant time 💯
The only exception to that criteria are runtime fields, but I don't think you have to deal with them on your side yet.